src/app/modules/gb-side-panel-module/accordion-components/gb-variables/gb-variables.component.ts

```diff
@@ -40,19 +39,28 @@ export class GbVariablesComponent implements OnInit {
         this.categorizedVariables.get(variable.type).push(variable);
       }
     });
+    this.updateCheckAllText();
   }
 
-  onCheckAll(e) {
-    this.checkAllText = e ? 'All selected' : 'None selected';
-    this.checkAll(e);
+  updateCheckAllText() {
+    let numSelected = 0;
+    this.categorizedVariables.forEach((list: Array<Concept>) => {
+      list.forEach((c: Concept) => {
+        if (c.selected) {
+          numSelected++;
+        }
+      });
+    });
+    this.checkAllText = `${numSelected} variables selected`;
```

Comment:
maybe variable(s)? to fit also numSelected = 1 :)

Reply:
good point!